src/ui/views/Board/components/Board/boardHelpers.ts

```diff
@@ -16,24 +16,23 @@ function getBasename(str: string) {
   return str.slice(lastSlash + 1);
 }
 
-const CHECKBOX_ITEM_REGEX = /^\s{0,3}-\s{1,4}\[.\]/;
-const COMPLETED_ITEM_REGEX = /^\s{0,3}-\s{1,4}\[\S\]/;
-
-// Look for markdown task lists in the record file content and count completion progress
-export async function getTaskProgress(recordId: string, app: App): Promise<string> {
+export async function getTaskProgress(
+  recordId: string,
+  app: App
+): Promise<string> {
   let progress = "";
 
   const file = app.vault.getFileByPath(recordId);
   if (file) {
-    const content = await app.vault.read(file);
-    const taskLines = content.split("\n").filter(l => CHECKBOX_ITEM_REGEX.test(l));
-
-    const totalTasks = taskLines.length;
-    const completedTasks = taskLines.map(l => COMPLETED_ITEM_REGEX.test(l)).filter(Boolean).length;
+    const totalTasks = app.metadataCache
+      .getFileCache(file)
+      ?.listItems?.filter((item) => item.task !== undefined);
 
     if (totalTasks) {
```

Comment:
totalTasks is an empty array in the present cache but no tasks case; empty arrays are `true` in js. Fixed

Reply:
Thanks!